RetailCoder.VBE/UI/CodeExplorer/Commands/IndentCommand.cs

```diff
@@ -88,14 +88,14 @@ protected override void OnExecute(object parameter)
             {
                 var declaration = ((ICodeExplorerDeclarationViewModel)node).Declaration;
 
-                var components = _state.AllUserDeclarations.Where(c => 
+                var componentDeclarations = _state.AllUserDeclarations.Where(c => 
                             c.DeclarationType.HasFlag(DeclarationType.Module) &&
                             c.Annotations.All(a => a.AnnotationType != AnnotationType.NoIndent) &&
-                            c.Project == declaration.Project);
+                            c.ProjectId == declaration.ProjectId);
 
-                foreach (var component in components)
+                foreach (var componentDeclaration in componentDeclarations)
                 {
-                    _indenter.Indent(component.QualifiedName.QualifiedModuleName.Component);
+                    _indenter.Indent(_state.ProjectsProvider.Component(componentDeclaration.QualifiedName.QualifiedModuleName));
```

Comment:
Seems an overload taking a `Declaration` could be nice here.

Reply:
An extension method might be ok. The interface itself cannot have such an overload without pulling it out of the `VBEditor` project.